packages/backend/server/src/base/job/queue/executor.ts

```diff
@@ -0,0 +1,73 @@
+import {
+  Injectable,
+  Logger,
+  OnModuleDestroy,
+  OnModuleInit,
+} from '@nestjs/common';
+import { Worker } from 'bullmq';
+
+import { Config } from '../../config';
+import { QueueRedis } from '../../redis';
+import { JOB_NAMESPACES } from './def';
+import { JobHandlerScanner } from './handler-scanner';
+
+@Injectable()
+export class JobExecutor implements OnModuleInit, OnModuleDestroy {
+  private readonly logger = new Logger('job');
+  private readonly workers: Record<string, Worker> = {};
+
+  constructor(
+    private readonly config: Config,
+    private readonly redis: QueueRedis,
+    private readonly scanner: JobHandlerScanner
+  ) {}
+
+  onModuleInit() {
+    if (AFFiNE.flavor.worker) {
+      this.startWorkers();
+    }
+  }
+
+  async onModuleDestroy() {
+    await this.stopWorkers();
+  }
+
+  async run(job: JobName, payload: any) {
+    const handler = this.scanner.getHandler(job);
+    if (!handler) {
+      this.logger.warn(`Job handler for [${job}] not found.`);
+      return;
+    }
+
+    try {
+      this.logger.debug(`Job started: ${job}`);
+      await handler(payload);
+      this.logger.debug(`Job finished: ${job}`);
+    } catch (e) {
+      this.logger.error(`Job failed: ${job}`, e);
+    }
+  }
+
+  private startWorkers() {
+    const queues = Array.from(JOB_NAMESPACES);
+
+    queues.forEach(queue => {
+      this.logger.verbose(`Queue Worker [${queue}] started.`);
+      this.workers[queue] = new Worker(
```

Comment:
should listen the worker `error` event to handler uncatch exception https://docs.bullmq.io/guide/workers